File: apps/android/app/src/main/kotlin/com/swab/android/carte/Labels.kt
package com.swab.android.carte

import com.swab.android.l10n.Fr
import com.swab.android.vault.VaultContact
import java.util.Locale

/**
 * Shared accessibility vocabulary for the carte (MAP-08 acceptance: every
 * contact announces name + ring, identically in map and list). Port of
 * apps/mobile/src/map/labels.ts. No Android imports.
 */
object Labels {
    val RING_LABEL: Map<Int, String> = mapOf(
        1 to Fr.RING_1,
        2 to Fr.RING_2,
        3 to Fr.RING_3,
        4 to Fr.RING_4,
    )

    /** « Léa — Très proche » for placed contacts, plain name otherwise. */
    fun contactLabel(contact: VaultContact): String {
        val ring = contact.ring
        return if (ring != null) "${contact.displayName} — ${RING_LABEL[ring]}" else contact.displayName
    }

    /** Up to two initials — glanceable node content, never the full name. */
    fun initials(displayName: String): String =
        displayName
            .split(Regex("\\s+"))
            .filter { it.isNotEmpty() }
            .take(2)
            .joinToString(separator = "") { part -> part.take(1).uppercase(Locale.FRENCH) }
}

File: apps/mobile/src/map/labels.ts
/**
 * Shared accessibility vocabulary for the carte (MAP-08 acceptance:
 * every contact announces name + ring, identically in map and list).
 */
import { t } from '../i18n/fr';
import type { IntimacyRing, VaultContact } from '../vault/vault';

export const RING_LABEL: Readonly<Record<IntimacyRing, string>> = {
  1: t('ring.1'),
  2: t('ring.2'),
  3: t('ring.3'),
  4: t('ring.4'),
};

/** « Léa — Très proche » for placed contacts, plain name otherwise. */
export function contactLabel(contact: VaultContact): string {
  return contact.ring !== undefined
    ? `${contact.displayName} — ${RING_LABEL[contact.ring]}`
    : contact.displayName;
}

/** Up to two initials — glanceable node content, never the full name. */
export function initials(displayName: string): string {
  return displayName
    .split(/\s+/u)
    .filter((part) => part.length > 0)
    .slice(0, 2)
    .map((part) => part[0]?.toLocaleUpperCase('fr') ?? '')
    .join('');
}
